airmozilla/main/views.py

```diff
@@ -1114,6 +1115,115 @@ def item_pubdate(self, event):
         return event.start_time
 
 
+def related_content(request, slug):
+    event = get_object_or_404(Event, slug=slug)
+
+    es = pyelasticsearch.ElasticSearch(settings.RELATED_CONTENT_URL)
+    es.refresh()
+
+    if request.user.is_active:
+        if is_contributor(request.user):
+            query = {
+                "query": {
+                    "filtered": {
+                        "query": {
+                            "more_like_this": {
+                                "fields": ["tags", "title", "channels"],
+                                "docs": [
+                                    {
+                                        "_index": "events",
+                                        "_type": "event",
+                                        "_id": event.id
+                                    }],
+                                "min_term_freq": 1,
+                                "max_query_terms": 5,
+                            }
+                        },
+                        "filter": {
+                            "must_not": {
+                                "term": {
+                                    "privacy": Event.PRIVACY_COMPANY
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        else:
+            query = {
+                "query": {
+                    "more_like_this": {
+                        "fields": ["tags", "title", "channels"],
+                        "docs": [
+                            {
+                                "_index": "events",
+                                "_type": "event",
+                                "_id": event.id
+                            }],
+                        "min_term_freq": 1,
+                        "max_query_terms": 5,
+                    }
+                }
+            }
+    else:
+        query = {
+            "query": {
+                "filtered": {
+                    "query": {
+                        "more_like_this": {
+                            "fields": ["tags", "title", "channels"],
+                            "docs": [
+                                {
+                                    "_index": "events",
+                                    "_type": "event",
+                                    "_id": event.id
+                                }],
+                            "min_term_freq": 1,
+                            "max_query_terms": 5,
+                        }
+                    },
+                    "filter": {
+                        "bool": {
+                            "must": {
+                                "term": {"privacy": Event.PRIVACY_PUBLIC}
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
See if you can refactor all of this so that it doesn't have to define the `"query"` part each time. The only thing that needs to change is the `"filter"` stuff.